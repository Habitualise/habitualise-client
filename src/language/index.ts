export const LABEL = {
  WELCOME_HEADER: 'Ready to\nbecome giga?',
  WELCOME_SUBHEADER: 'Start your journey with us\ntoday on project habits.',
  SIGN_UP: 'Sign Up',
  ALREADY_HAVE_ACCOUNT: 'I already have an account',
  LOGIN_SCREEN: 'Login Screen',
  LOG_IN: 'Log In',
  LOG_OUT: 'Log Out',
  WELCOME_SCREEN: 'Welcome Screen',
  TODAY_SCREEN: 'Today Screen',
  TODAY_HEADER: "Today's Habits",
  TODAY: 'Today',
  HABITS_SCREEN: 'Habits Screen',
  HABITS_HEADER: 'My Habits',
  ACTIVE: 'Active',
  ARCHIVED: 'Archived',
  HABITS: 'Habits',
  GROUPS_SCREEN: 'Groups Screen',
  GROUPS: 'Groups',
  SETTINGS_SCREEN: 'Settings Screen',
  SETTINGS: 'Settings',
  LOGGED_IN_AS: (username: string) => `Logged in as ${username}`,
  DAYS_COMPLETED: (percentage: number) => `${percentage}% of days completed`,
  ADD_HABIT: 'Add a new habit',
  HABIT_NAME: 'Habit Name',
  REPEAT: 'REPEAT',
};
