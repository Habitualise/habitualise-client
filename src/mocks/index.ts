export const mockUser = {
  email: 'lokchoy@gmail.com',
  email_verified: true,
  sub: 'auth0|5f2e6f7e6e1c6f0c6c0d6b7d',
};

export const mockHabits = [
  {
    id: 1,
    name: 'Running',
    iconName: 'run',
    colour: 'rgb(66, 135, 245)',
    isCompleted: false,
    completionPercentage: 67,
    status: 'active',
    daysDue: [1, 2, 3, 4, 5, 6],
    lastSevenCompletions: ['2023-02-01', '2023-02-02', '2023-02-03'],
  },
  {
    id: 2,
    name: 'Yoga',
    iconName: 'yoga',
    colour: 'rgb(245, 179, 66)',
    isCompleted: false,
    completionPercentage: 0,
    status: 'active',
    daysDue: [1, 2, 3, 4],
    lastSevenCompletions: [],
  },
  {
    id: 3,
    name: 'Meditation',
    iconName: 'meditation',
    colour: 'rgb(207, 65, 250)',
    isCompleted: false,
    completionPercentage: 99,
    status: 'active',
    daysDue: [0, 1, 2, 3, 4, 5, 6],
    lastSevenCompletions: [
      '2023-02-01',
      '2023-02-02',
      '2023-02-07',
      '2023-02-13',
      '2023-02-15',
      '2023-02-16',
    ],
  },
];
