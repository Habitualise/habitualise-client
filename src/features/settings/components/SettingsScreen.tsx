import React from 'react';
import {StyleSheet} from 'react-native';
import {Button, Text} from 'react-native-paper';
import {useAuth0} from 'react-native-auth0';

import {commonStyles} from '@/components/styles';
import {LABEL} from '@/language';
import {axios} from '@/lib/axios';
import {useStore, ACTIONS} from '@/context/StoreContext';
import PaperView from '@/components/PaperView';
import {SafeAreaView} from 'react-native-safe-area-context';

export const SettingsScreen = () => {
  const {clearSession, user} = useAuth0();
  const {dispatch} = useStore();

  const logOut = async () => {
    try {
      await clearSession();
      dispatch({type: ACTIONS.HANDLE_LOGOUT});
    } catch (e) {
      console.log(e);
    }
  };

  const testAxiosHealth = async () => {
    try {
      const response = await axios.get('/api/health');
      console.log('=== response from TodayScreen.tsx [19] ===', response);
    } catch (error) {
      console.log('=== error from TodayScreen.tsx [21] ===', error);
    }
  };

  return (
    <SafeAreaView
      style={{
        flex: 1,
      }}
      edges={['top', 'left', 'right']}>
      <PaperView style={commonStyles.container}>
        <Text variant="displaySmall" style={styles.header}>
          {LABEL.SETTINGS_SCREEN}
        </Text>
        <Text variant="labelSmall">{LABEL.LOGGED_IN_AS(user?.name)}</Text>
        <Button mode="contained" onPress={logOut} style={styles.logOutButton}>
          {LABEL.LOG_OUT}
        </Button>
        <Button onPress={testAxiosHealth}>Test Axios Health</Button>
      </PaperView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  logOutButton: {
    marginTop: 20,
  },
  header: {
    marginBottom: 20,
  },
});
