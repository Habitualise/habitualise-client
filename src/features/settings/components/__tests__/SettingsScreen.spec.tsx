import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react-native';

import { SettingsScreen } from '../SettingsScreen';
import { LABEL } from '@/language';
import { mockUser } from '@/mocks';

const mockClearSession = jest.fn();

jest.mock('react-native-auth0', () => ({
  useAuth0: () => ({
    clearSession: mockClearSession,
    user: mockUser,
  }),
}));

describe('SettingsScreen', () => {
  it('should match snapshot', () => {
    const { toJSON } = render(<SettingsScreen />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should call useAuth0 clearSession function when sign out button is pressed', () => {
    render(<SettingsScreen />);

    fireEvent.press(screen.getByText(LABEL.LOG_OUT));
    expect(mockClearSession).toHaveBeenCalled();
  });

  //  not working right now
  //   it('should display formatted user name label', async () => {
  //     const wrapper = await render(<SettingsScreen />);

  //     console.log(`=====================${mockUser.email}=====================`);

  //     await new Promise((r) => setTimeout(r, 1000));
  //     wrapper.update(<SettingsScreen />);
  //     wrapper.debug();

  //     await expect(
  //       screen.getByText(LABEL.LOGGED_IN_AS(mockUser.email)),
  //     ).toBeTruthy();
  //   });
});
