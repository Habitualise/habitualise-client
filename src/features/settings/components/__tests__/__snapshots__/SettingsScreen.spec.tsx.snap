// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsScreen should match snapshot 1`] = `
<RNCSafeAreaView
  edges={
    [
      "top",
      "left",
      "right",
    ]
  }
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(255, 251, 254, 1)",
        },
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "paddingHorizontal": 15,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontSize": 36,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 44,
          },
          {
            "textAlign": "left",
          },
          {
            "writingDirection": "ltr",
          },
          {
            "marginBottom": 20,
          },
        ]
      }
    >
      Settings Screen
    </Text>
    <Text
      style={
        [
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontSize": 11,
            "fontWeight": "500",
            "letterSpacing": 0.5,
            "lineHeight": 16,
          },
          {
            "textAlign": "left",
          },
          {
            "writingDirection": "ltr",
          },
          undefined,
        ]
      }
    >
      Logged in as undefined
    </Text>
    <View
      collapsable={false}
      style={
        {
          "alignSelf": undefined,
          "bottom": undefined,
          "end": undefined,
          "left": undefined,
          "position": undefined,
          "right": undefined,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "start": undefined,
          "top": undefined,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "rgba(103, 80, 164, 1)",
              "borderColor": "transparent",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 0,
              "marginTop": 20,
              "minWidth": 64,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                false,
                {
                  "borderRadius": 20,
                },
              ]
            }
            testID="button"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  [
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      {
                        "color": "rgba(255, 255, 255, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Log Out
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        {
          "alignSelf": undefined,
          "bottom": undefined,
          "end": undefined,
          "left": undefined,
          "position": undefined,
          "right": undefined,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "start": undefined,
          "top": undefined,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 0,
              "minWidth": 64,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                false,
                {
                  "borderRadius": 20,
                },
              ]
            }
            testID="button"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  [
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 12,
                      },
                      undefined,
                      false,
                      {
                        "color": "rgba(103, 80, 164, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Test Axios Health
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
